packages/feedback-components/src/feedback/text-visualizer.ts

```diff
@@ -78,95 +82,199 @@ export function FeedbackText(props: FeedbackTextProps) {
+  const selection = window.getSelection();
+  if (!selection || selection.isCollapsed || selection.rangeCount === 0 || !container) return null;
 
-        // check if blank
-        if (payload.text === " ") {
-          console.log("Blank tag found, ignoring");
-          return;
-        }
+  const range = selection.getRangeAt(0);
 
-        // check if duplicate
-        const duplicate = tags.find(
-          (tag) => tag.start === payload.start && tag.end === payload.end - 1
-        );
+  if (!container.contains(range.startContainer) || !container.contains(range.endContainer)) {
+    return null;
+  }
 
-        if (duplicate) {
-          console.log("Duplicate tag found, ignoring");
-          return;
-        }
+  const preRange = document.createRange();
+  preRange.setStart(container, 0);
+  preRange.setEnd(range.startContainer, range.startOffset);
+  const start = preRange.toString().length;
 
-        // remove ending whitespace if needed
-        if( payload.text.endsWith(" ")) {
-          payload.text = payload.text.slice(0, -1);
-          payload.end -= 1;
-        }
+  const selectedText = range.toString();
+  const end = start + selectedText.length;
 
-        const overlap = tags.some(
-            (tag) =>
-              tag.start <= payload.start &&
-              tag.end >= payload.end &&
-              tag.id !== undefined);
+  return {
+    start,
+    end,
+    text: selectedText
+  };
+}
 
-        // check if inside
-        if (overlap && !allowOverlap) {
-          console.log("Tag is inside another tag, ignoring");
-          return;
-        }
+function addTag({ tag, dispatch, text, allTags, allowOverlap }) {
+  const { start, end } = tag;
+  let payload = { start, end, text: text.slice(start, end) };
 
-        dispatch({ type: "create-node", payload });
-        return;
-      }
+  if (payload.text.trim() === "") {
+    console.log("Blank tag found, ignoring");
+    return;
+  }
+
+  const duplicate = allTags.find(
+    (t) => t.start === payload.start && t.end === payload.end
+  );
+  if (duplicate) {
+    console.log("Duplicate tag found, ignoring");
+    return;
+  }
+
+  if (payload.text.endsWith(" ")) {
+    payload.text = payload.text.slice(0, -1);
+    payload.end -= 1;
+  }
+
+  const overlap = allTags.some(
+    (t) =>
+      t.start <= payload.start &&
+      t.end >= payload.end &&
+      t.id !== undefined
+  );
 
-      // allow nested tags to be clicked
-      if (!clicked) {
-        clicked = true;
+  if (overlap && !allowOverlap) {
+    console.log("Tag is inside another tag, ignoring");
+    return;
+  }
 
-        const tagIDs = new Set(tags.map((d) => d.id));
-        const removedIds = allTags.map((d) => d.id).filter((d) => !tagIDs.has(d));
+  dispatch({ type: "create-node", payload });
+}
 
-        if (removedIds.length > 0) {
-          dispatch({
-            type: "toggle-node-selected",
-            payload: { ids: removedIds },
-          });
+export function HighlightedText(props: {
```

Comment:
Can you use or modify the highlighted text component [here](https://github.com/UW-Macrostrat/web-components/blob/main/packages/feedback-components/src/extractions/index.ts) rather than creating a new one? Or at least make it use similar inputs?

Reply:
Actually, never mind you're already doing this! My mistake.